sdk/frc42dispatch/hash_method.go

```diff
@@ -0,0 +1,86 @@
+// Package frc42dispatch implement frc42 reference https://github.com/filecoin-project/FIPs/blob/master/FRCs/frc-0042.md
+package frc42dispatch
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+
+	"github.com/minio/blake2b-simd"
+
+	"github.com/filecoin-project/go-state-types/abi"
+
+	"golang.org/x/exp/utf8string"
+)
+
+// CONSTRUCTORMETHODNAME default constructor name
+const CONSTRUCTORMETHODNAME = "Constructor"
+
+// CONSTRUCTORMETHODNUMBER default constructor method number
+const CONSTRUCTORMETHODNUMBER abi.MethodNum = 1
+
+// DIGESTCHUNKLENGTH chunk to generate u32 method number
+const DIGESTCHUNKLENGTH = 4
+
+// FIRSTMETHODNUMBER Method numbers below FIRST_METHOD_NUMBER are reserved for other use
+const FIRSTMETHODNUMBER = 1 << 24
+
+// GenMethodNumber generate method number by method name
+func GenMethodNumber(name string) (abi.MethodNum, error) {
+	err := checkMethodName(name)
+	if err != nil {
+		return 0, err
+	}
+	if name == CONSTRUCTORMETHODNAME {
+		return CONSTRUCTORMETHODNUMBER, nil
+	}
+	methodName := fmt.Sprintf("1|%s", name)
+	hasher := blake2b.New512()
+	_, err = hasher.Write([]byte(methodName))
+	if err != nil {
+		return 0, err
+	}
+	methodHashBytes := hasher.Sum(nil)
+	var first, end = 0, 1
+	for i := 1; i <= len(methodHashBytes); i++ {
+		if i%DIGESTCHUNKLENGTH == 0 {
+			methodId := binary.BigEndian.Uint32(methodHashBytes[first:end]) //nolint
+			if methodId >= FIRSTMETHODNUMBER {
+				return abi.MethodNum(methodId), nil
+			}
+		}
+		end++
+	}
```

Comment:
This chunking doesn't look quite right to me. E.g. the `first` variable is never updated. I think `i` and `end` are unnecessary, but `first` should be incremented by DIGESTCHUNKLENGTH each time round the loop.

Reply:
thanks, i have fix this. and shall i make hasher configrable like rust?, will this hash function change in the future?

Reply:
It's unlikely to change. The parameterisation in Rust was mainly so that it can be outsourced to a VM syscall from within actors.